File: GeneratedEntityReaderClasses/Components/InventoryComponent.cs
namespace GeneratedNoitaClasses;
public class InventoryComponent : NoitaComponentBase,  iNoitaType<InventoryComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt ui_container_type { get; set; } = new();
	public NoitaCVector2<NoitaInt> ui_container_size { get; set; } = new();
	public NoitaCVector2<NoitaInt> ui_element_size { get; set; } = new();
	public NoitaCVector2<NoitaInt> ui_position_on_screen { get; set; } = new();
	public NoitaString ui_element_sprite { get; set; } = new();
	public NoitaString actions { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		ui_container_type.Read(s);
		ui_container_size.Read(s);
		ui_element_size.Read(s);
		ui_position_on_screen.Read(s);
		ui_element_sprite.Read(s);
		actions.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		ui_container_type.Write(s);
		ui_container_size.Write(s);
		ui_element_size.Write(s);
		ui_position_on_screen.Write(s);
		ui_element_sprite.Write(s);
		actions.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/InventoryComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class InventoryComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    ui_container_type:NoitaInt = field(default_factory=NoitaInt)
    ui_container_size:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaInt))
    ui_element_size:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaInt))
    ui_position_on_screen:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaInt))
    ui_element_sprite:NoitaString = field(default_factory=NoitaString)
    actions:NoitaString = field(default_factory=NoitaString)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.ui_container_type.Read(s)
        self.ui_container_size.Read(s)
        self.ui_element_size.Read(s)
        self.ui_position_on_screen.Read(s)
        self.ui_element_sprite.Read(s)
        self.actions.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.ui_container_type.Write(s)
        self.ui_container_size.Write(s)
        self.ui_element_size.Write(s)
        self.ui_position_on_screen.Write(s)
        self.ui_element_sprite.Write(s)
        self.actions.Write(s)

